### PR TITLE
Define Parent interface and ElementFactory

### LINKED ISSUE #11: Define Parent interface for Element

### PR DESCRIPTION
**Pull request by pavelanihimovsky**
resolves #11

### CODE REVIEW

File: aquality_selenium_core/elements/parent.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+T = TypeVar('T')
+
+
+class Parent(ABC):
+    """
+    Defines behavior of element with child elements.
+    """
+
+    @abstractmethod
+    def find_child_element(self, child_locator: By, name: str = None,
```
Comment (mialeska) [inline aquality_selenium_core/elements/parent.py:18]: please add fing_child_elements method into this interface

File: aquality_selenium_core/elements/element.py [hunk 2]
```diff
@@ -154,3 +158,45 @@ def _cache(self) -> AbstractElementCacheHandler:
...
+
+    def find_child_element(
+        self,
+        child_locator: By,
+        name: str,
+        supplier: Callable[[By, str, ElementState], T],
+        state: ElementState = ElementState.DISPLAYED,
+    ) -> T:
+        """
+        Find child element ot type T of current element by its locator.
```
Comment (n-verbitsky) [inline aquality_selenium_core/elements/element.py:170]: Maybe 'Find the child element **of** type T of the current element by its locator.' ?

File: aquality_selenium_core/elements/element_factory.py [hunk 3]
```diff
@@ -1,8 +1,100 @@
...
+        expected_count: ElementsCount = ElementsCount.ANY,
+        state: ElementState = ElementState.DISPLAYED,
+    ) -> T:
+        """
+        Find list of elements by base locator.
+
+        :param locator: Base elements locator.
+        :param name: Elements name.
+        :param supplier: Delegate that defines constructor of element in case of custom elements.
+        :param expected_count: Expected number of elements that have to be found (zero, more then zero, any).
```
Comment (n-verbitsky) [inline aquality_selenium_core/elements/element_factory.py:96]: Maybe '(zero, more **than** zero, any).'?

File: aquality_selenium_core/elements/elements_count.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+"""Module defines enumeration for element counts."""
+from enum import Enum
+
+
+class ElementsCount(Enum):
+    """Enumeration with possible count of elements."""
+
+    ZERO = 0
+    MORE_THEN_ZERO = 1
```
Comment (n-verbitsky) [inline aquality_selenium_core/elements/elements_count.py:9]: Seemingly typo in the word  **THAN**

File: aquality_selenium_core/elements/parent.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        expected_count: ElementsCount = ElementsCount.ANY,
+        state: ElementState = ElementState.DISPLAYED,
+    ) -> List[T]:
+        """
+        Find child elements of current element by its locator.
+
+        :param child_locator: Locator of child elements relative to their parent.
+        :param name: Child elements name.
+        :param supplier: Delegate that defines constructor of child element in case of custom element type.
+        :param expected_count: Expected number of elements that have to be found (zero, more then zero, any).
```
Comment (n-verbitsky) [inline aquality_selenium_core/elements/parent.py:53]: Typo in word **than**

**Pull request merged**